### DISCUSSION TITLE
Water and lava block blast resistance overrides should apply to their respective fluids too

### DISCUSSION THREAD

**Discussion by galacticwarrior9** [PurpurMC/Purpur#1244] [__Completed__] [answered]
Not sure whether this should be classified as a bug report or feature request but I'll go with the latter. 

Right now, lowering the blast resistance of water and lava blocks in the `blast-resistance-overrides` section of the Purpur config has no effect. That's because there are two sets of blast resistances affecting them: 

- the value returned by `Block#getExplosionResistance`, which the `blast-resistance-overrides` config changes
- the value returned by `Fluid#getExplosionResistance`, which is hardcoded to be 100 in the `WaterFluid` and `LavaFluid` classes

```java
    @Override
    protected float getExplosionResistance() {
        return 100.0F;
    }
```

In `ExplosionDamageCalculator#getBlockExplosionResistance`, we see that the effective blast resistance of these blocks will always be the highest of those two values, i.e. it can never be lower than 100. 

```java
    public Optional<Float> getBlockExplosionResistance(Explosion explosion, BlockGetter world, BlockPos pos, BlockState blockState, FluidState fluidState) {
        return blockState.isAir() && fluidState.isEmpty() ? Optional.empty() : Optional.of(Math.max(blockState.getBlock().getExplosionResistance(), fluidState.getExplosionResistance()));
    }
```

So the problem is that there is no way to reduce the blast resistance of water and lava using the current `blast-resistance-overrides` config. It would make more sense if the config also changed the value returned by `Fluid#getExplosionResistance`.

There are several reasons why a server could want to do this. In my case, I would like underwater explosions to work. The fact that they do not work is currently a significant problem on my server since it revolves around [naval combat](https://youtu.be/OdkIa2qLB_0).

**Comment by granny** [comment, marked as answer]
implemented as of https://github.com/PurpurMC/Purpur/commit/789233aa7f164c44e38b15958e865e54d2de7eb1 on build #1906
